saleor/core/utils/random_data.py

```diff
@@ -920,20 +915,23 @@ def _create_staff_user(staff_password, email=None, superuser=False):
     if not email:
         email = get_email(first_name, last_name)
 
-    staff_user = User.objects.create_user(
-        first_name=first_name,
-        last_name=last_name,
-        email=email,
-        password=staff_password,
-        default_billing_address=address,
-        default_shipping_address=address,
-        is_staff=True,
-        is_active=True,
-        is_superuser=superuser,
-        search_document=_prepare_search_document_value(
-            User(email=email, first_name=first_name, last_name=last_name), address
-        ),
-    )
+    staff_user = User.objects.filter(email=email).first()
```

Comment:
Not sure if I understand. There is the same query at the beginning of this function. That call `return` in case when user already exists. Do we need that part in that case? In this moment `staff_user` should be always `None`. (or maybe I am missing something 😄 )

Reply:
Yes, there's an `if` in between when email is not provided. Previously, it was breaking when there already was a user in DB for the newly generated email.

Reply:
So it seems like instead of looking for an existing user two times, we can just leave the second query and drop the query and check if the user exists from lines 909 to 911.